mindmeld/auto_annotator.py

```diff
@@ -335,125 +320,126 @@ def _item_to_query_entity(item, processed_query):
     For more information on the supported entities for the Spacy Annotator check the MindMeld docs.
     """
 
-    def __init__(self, app_path, config=None):
-        """ Initializes an annotator.
+    def __init__(self, *args, **kwargs):
+        """Initializes a SpacyAnnotator.
 
         Args:
-            app_path (str): The location of the MindMeld app
-            config (dict, optional): A config object to use. This will
-                override the config specified by the app's config.py file.
+            app_path (str): The location of the MindMeld app.
+            annotation_rules (list): List of Annotation rules.
+            language (str, optional): Language as specified using a 639-1/2 code.
+            locale (str, optional): The locale representing the ISO 639-1 language code and \
+                ISO3166 alpha 2 country code separated by an underscore character.
+            overwrite (bool): Whether to overwrite existing annotations with conflicting spans.
+            spacy_model_size (str, optional): Size of the Spacy model to use. ("sm", "md", or "lg")
+            unannotate_supported_entities_only (bool): Only allow removal of supported entities.
+            unannotation_rules (list): List of Annotation rules.
         """
-        super().__init__(app_path=app_path, config=config)
+        super().__init__(*args, **kwargs)
+        if self.language not in SPACY_ANNOTATOR_SUPPORTED_LANGUAGES:
+            raise ValueError(
+                "Spacy does not currently support: {!r}.".format(self.language)
+            )
+        self.spacy_model_size = kwargs.get("spacy_model_size", "lg")
+        if self.spacy_model_size not in SPACY_ANNOTATOR_MODEL_SIZES:
+            raise ValueError(
+                "{!r} is not a valid model size. Select from: {!r}.".format(
+                    self.language, " ".join(SPACY_ANNOTATOR_MODEL_SIZES)
+                )
+            )
+        self.nlp = self._load_model()
 
-        self.model = self.config.get("spacy_model", EN_CORE_WEB_LG)
-        self.nlp = SpacyAnnotator._load_model(self.model)
-        self.duckling = DucklingRecognizer.get_instance()
-        self.ANNOTATOR_TO_DUCKLING_ENTITY_MAPPINGS = {
-            "money": "sys_amount-of-money",
-            "cardinal": "sys_number",
-            "ordinal": "sys_ordinal",
-            "person": "sys_person",
-            "percent": "sys_percent",
-            "distance": "sys_distance",
-            "quantity": "sys_weight",
-        }
+    def _get_spacy_model_name(self):
+        """Get the name of a Spacy Model.
 
-    @staticmethod
-    def _load_model(model):
-        """ Load Spacy English model. Download if needed.
+        Returns:
+            spacy_model_name (str): Name of the Spacy NER model
+        """
+        model_type = "web" if self.language in SPACY_ANNOTATOR_WEB_LANGUAGES else "news"
+        return "_".join([self.language, "core", model_type, self.spacy_model_size])
```

Comment:
Nit: It might be clearer to do a fstring here: `f"{self.language}-{core}-..."`

Reply:
Done